### DISCUSSION TITLE
Pico requires Arduino mbed core

### DISCUSSION THREAD

**Discussion by w9ran** [tttapa/Control-Surface#999] [Q&A] [answered]
This is just a note for anyone else who wishes to use this excellent library on the Rasperry Pi Pico.    The good news is, Midi over USB works very well on the Pico, but there are two cores that allow programming the Pic on the Arduino IDE.   The Arduino Mbed core works and is the one that should be used.    However many Pico fans are using the Earle Philhower core which supports many Pico devices and has additional features - but it doesn't work with Control-Surface.    The examples will compile without errors and appear to run (i.e. something comes out the USB port when devices attached to the Pico area actuated) - but the Pico is not recognized as a Midi device.  Simple selecting Raspberry Pi Pico under Arduino Mbed OS RP2040 boards and recompiling is all that's needed.

Bob

**Comment by tttapa** [comment, marked as answer]
MIDI over USB is supported using the Earle Philhower core as well, but you have to select the Adafruit TinyUSB USB stack.

See also <https://tttapa.github.io/Control-Surface-doc/Doxygen/d8/d4a/md_pages_MIDI-over-USB.html#mbed-os-boards-with-native-usb-support>.

**Reply by w9ran** [reply]
Thanks - I missed that, but it's always good to have choices.  Especially given the wider range of platforms that the Philhower core supports.
